src/Symfony/Component/DependencyInjection/EnvVarProcessor.php

```diff
@@ -67,19 +68,39 @@ public function getEnv($prefix, $name, \Closure $getEnv)
             if (!\is_array($array)) {
                 throw new RuntimeException(sprintf('Resolved value of "%s" did not result in an array value.', $next));
             }
-            if (!array_key_exists($key, $array)) {
-                throw new RuntimeException(sprintf('Key "%s" not found in "%s" (resolved from "%s")', $key, json_encode($array), $next));
+
+            if (!isset($array[$key]) && !array_key_exists($key, $array)) {
+                throw new EnvNotFoundException(sprintf('Key "%s" not found in "%s" (resolved from "%s")', $key, json_encode($array), $next));
             }
 
             return $array[$key];
         }
 
+        if ('default' === $prefix) {
+            if (false === $i) {
+                throw new RuntimeException(sprintf('Invalid configuration: env var "default:%s" does not contain a default value.', $name));
+            }
+
+            $next = substr($name, $i + 1);
+            $default = substr($name, 0, $i);
+
+            try {
+                return $getEnv($next);
+            } catch (EnvNotFoundException $e) {
+                if (!$this->container->hasParameter($default)) {
+                    throw new EnvNotFoundException(sprintf('Parameter "%s" not found. (fallback from not found "%s")', $default, $next), 0, $e);
```

Comment:
`sprintf('Invalid env fallback in "default:%s": parameter "%s" not found.', $name, $default)`?

Reply:
should it be a runtime exception btw?

Reply:
fix message.
what's about `default:my_fallback:default:optionnal_parameter_injected_by_third_party:foo`?